components/ws-manager/pkg/manager/status.go

```diff
@@ -800,7 +808,24 @@ func (m *Manager) isWorkspaceTimedOut(wso workspaceObjects) (reason string, err
 			// the workspace is up and running, but the user has never produced any activity
 			return decide(start, m.Config.Timeouts.TotalStartup, activityNone)
 		} else if isClosed {
-			return decide(*lastActivity, m.Config.Timeouts.AfterClose, activityClosed)
+			reason, err := func() (string, error) {
+				afterClosed := m.Config.Timeouts.AfterClose
+				if ctv, ok := wso.Pod.Annotations[customClosedTimeoutAnnotation]; ok {
+					if ct, err := time.ParseDuration(ctv); err == nil {
+						// skip check if closed timeout set to 0
+						if ct == 0 {
+							return "", nil
+						}
+						afterClosed = util.Duration(ct)
+					} else {
+						log.WithError(err).WithField("customClosedTimeout", ctv).WithFields(wsk8s.GetOWIFromObject(&wso.Pod.ObjectMeta)).Warn("pod had custom closed timeout annotation set, but could not parse its value. Defaulting to ws-manager config.")
+					}
+				}
+				return decide(*lastActivity, afterClosed, activityClosed)
+			}()
+			if reason != "" {
+				return reason, err
+			}
```

Comment:
why no longer return if `reason == ""`? This is a logic change from before, wondering what's the reason.

Specifically, `decide()` could also return `"", nil`, do we want to continue (not return) if that happens?